pkg/controller/route53resolver/resolverruleassociation/controller.go

```diff
@@ -0,0 +1,166 @@
+	errDelete       = "failed to delete the Resolver Rule Association resource"
+	errStatusUpdate = "cannot update status of Resolver Rule Association custom resource"
+)
+
+// SetupResolverRuleAssociation adds a controller that reconciles ResolverRuleAssociation
+func SetupResolverRuleAssociation(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.ResolverRuleAssociationKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.ResolverRuleAssociation{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.ResolverRuleAssociationGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.ResolverRuleAssociation)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: awsr53r.New(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client r53r.ResolverRuleAssociationClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.ResolverRuleAssociation)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+	var response *awsr53r.GetResolverRuleAssociationResponse
+	var err error
+
+	response, err = e.client.GetResolverRuleAssociationRequest(&awsr53r.GetResolverRuleAssociationInput{
+		ResolverRuleAssociationId: aws.String(meta.GetExternalName(cr))}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(r53r.IsResolverRuleAssociationNotFoundErr, err), errDescribe)
+	}
+
+	observed := response.ResolverRuleAssociation
+	current := cr.Spec.ForProvider.DeepCopy()
+
+	cr.SetConditions(xpv1.Available())
+
+	cr.Status.AtProvider = r53r.GenerateRoute53ResolverObservation(*observed)
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        true,
+		ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider),
```

Comment:
```suggestion
```
It seems we don't do any late initialization, so it should be fine to return false here.